pkg/acsengine/engine.go

```diff
@@ -560,6 +560,8 @@ func getParameters(cs *api.ContainerService, isClassicMode bool) (paramsMap, err
 		addValue(parametersMap, "cloudProviderRatelimitBucket", strconv.Itoa(properties.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket))
 		addValue(parametersMap, "kubeClusterCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
 		addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.NonMasqueradeCidr)
+		addValue(parametersMap, "generatorCode", DefaultGeneratorCode)
```

Comment:
The service calls GenerateTemplate() and passes `api.ContainerService` (and gets back template & params) so the info what `generatorCode` and `orchestratorName` can be derived from `api.ContainerService`. That way we prevent 2 code paths how many parameters are set.